02_cancer_type_classification/nbconverted/lasso_range_gene.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments
+# 
+# We want to see whether smaller models (i.e. models with fewer nonzero features) tend to generalize to new cancer types better than larger ones; this script compares/visualizes those results.
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_valid'
+)
+
+training_dataset = 'all_other_cancers'
+results_dir = os.path.join(base_results_dir, training_dataset)
+
+plot_gene = 'ATRX'
+metric = 'aupr'
+
+output_plots = True
+output_plots_dir = cfg.cancer_type_lasso_range_dir
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+nz_coefs_df = []
+
+# get pancancer coefs info for now
```

Comment:
Do you want to change this in the future? Maybe throw a TODO on?

Reply:
Ahh, this comment isn't accurate anymore since I have the `training_dataset` variable in the previous cell which can be used to select the dataset now. I'll get rid of it.